Add some missing search related tests

nikomakela opened the pull request:
LIIKUNTA-358

Add unit tests for 
- Venue's similar venues, 
- upcoming events
- combined search

Browser tests
- refactor and move the cookie consent and the user role to common test package
- test the navigation between the search page and event details page

### How to test

Unit tests in app folder with `yarn test` or globally with `yarn g:test`.

Browser tests by 

1. first building the app with `yarn build`, 
2. then running the built version with `yarn start` 
3. running the browser tests in terminal with `yarn test:browser:ci` or in a browser window with `yarn test:browser`

nikomakela committed:
test(sports): upcoming events of a venue

terovirtanen commented:
## TestCafe result is __success__ for https://liikunta-pr205.dev.hel.ninja :laughing::tada::tada::tada:

nikomakela committed:
test(sports): search utilities

terovirtanen commented:
**Events-Helsinki branch is deployed to platta: https://tapahtumat-pr205.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**Hobbies-Helsinki branch is deployed to platta: https://harrastukset-pr205.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**Sports-Helsinki branch is deployed to platta: https://liikunta-pr205.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
## TestCafe result is __success__ for https://tapahtumat-pr205.dev.hel.ninja :laughing::tada::tada::tada:

terovirtanen commented:
## TestCafe result is __success__ for https://harrastukset-pr205.dev.hel.ninja :laughing::tada::tada::tada:

terovirtanen commented:
## TestCafe result is __success__ for https://liikunta-pr205.dev.hel.ninja :laughing::tada::tada::tada:

terovirtanen commented:
**Hobbies-Helsinki branch is deployed to platta: https://harrastukset-pr205.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**Events-Helsinki branch is deployed to platta: https://tapahtumat-pr205.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**Hobbies-Helsinki branch is deployed to platta: https://harrastukset-pr205.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**Sports-Helsinki branch is deployed to platta: https://liikunta-pr205.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**$(upper(replace(parameters.proxy,'-', ' '))) is deployed to platta: https://events-graphql-proxy-pr205.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
## TestCafe result is __success__ for https://tapahtumat-pr205.dev.hel.ninja :laughing::tada::tada::tada:

terovirtanen commented:
## TestCafe result is __success__ for https://liikunta-pr205.dev.hel.ninja :laughing::tada::tada::tada:

terovirtanen commented:
**$(upper(replace(parameters.proxy,'-', ' '))) is deployed to platta: https://venue-graphql-proxy-pr205.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**Sports-Helsinki branch is deployed to platta: https://liikunta-pr205.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**Events-Helsinki branch is deployed to platta: https://tapahtumat-pr205.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
## TestCafe result is __success__ for https://liikunta-pr205.dev.hel.ninja :laughing::tada::tada::tada:

terovirtanen commented:
**Hobbies-Helsinki branch is deployed to platta: https://harrastukset-pr205.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
## TestCafe result is __success__ for https://tapahtumat-pr205.dev.hel.ninja :laughing::tada::tada::tada:

terovirtanen commented:
## TestCafe result is __success__ for https://harrastukset-pr205.dev.hel.ninja :laughing::tada::tada::tada:

terovirtanen commented:
**VENUE GRAPHQL PROXY is deployed to platta: https://venue-graphql-proxy-pr205.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**EVENTS GRAPHQL PROXY is deployed to platta: https://events-graphql-proxy-pr205.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**Sports-Helsinki branch is deployed to platta: https://liikunta-pr205.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**Events-Helsinki branch is deployed to platta: https://tapahtumat-pr205.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**Hobbies-Helsinki branch is deployed to platta: https://harrastukset-pr205.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**EVENTS GRAPHQL PROXY is deployed to platta: https://events-graphql-proxy-pr205.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**VENUE GRAPHQL PROXY is deployed to platta: https://venue-graphql-proxy-pr205.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
## TestCafe result is __failed__ for https://tapahtumat-pr205.dev.hel.ninja :crying_cat_face::anger::boom::boom:

terovirtanen commented:
## TestCafe result is __failed__ for https://harrastukset-pr205.dev.hel.ninja :crying_cat_face::anger::boom::boom:

nikomakela committed:
chore: update the Testcafe

nikomakela committed:
refactor: browser tests consent modal and role usage

The user role was used wrongly, because the page set to the user role
and a page set to the fixture made the whole test jump between the pages
or reload the page, but also made the user role cookies to be cleared.

The cookie consent is now refactored and moved fully to the common package.
Now the app does not need to implement it, which means less code to maintain.

terovirtanen commented:
**Events-Helsinki branch is deployed to platta: https://tapahtumat-pr205.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**Hobbies-Helsinki branch is deployed to platta: https://harrastukset-pr205.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**VENUE GRAPHQL PROXY is deployed to platta: https://venue-graphql-proxy-pr205.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**Sports-Helsinki branch is deployed to platta: https://liikunta-pr205.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**Hobbies-Helsinki branch is deployed to platta: https://harrastukset-pr205.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**VENUE GRAPHQL PROXY is deployed to platta: https://venue-graphql-proxy-pr205.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**Events-Helsinki branch is deployed to platta: https://tapahtumat-pr205.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**Sports-Helsinki branch is deployed to platta: https://liikunta-pr205.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**EVENTS GRAPHQL PROXY is deployed to platta: https://events-graphql-proxy-pr205.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
## TestCafe result is __success__ for https://harrastukset-pr205.dev.hel.ninja :laughing::tada::tada::tada:

terovirtanen commented:
## TestCafe result is __failed__ for https://liikunta-pr205.dev.hel.ninja :crying_cat_face::anger::boom::boom:

terovirtanen commented:
## TestCafe result is __failed__ for https://liikunta-pr205.dev.hel.ninja :crying_cat_face::anger::boom::boom:

terovirtanen commented:
## TestCafe result is __failed__ for https://tapahtumat-pr205.dev.hel.ninja :crying_cat_face::anger::boom::boom:

nikomakela commented on packages/common-tests/browser-tests/roles/allCookiesUser.ts:
```diff
@@ -0,0 +1,22 @@
+} from 'events-helsinki-common-tests/browser-tests';
+import { Role } from 'testcafe';
+
+export const acceptAllCookies = async () => {
+  console.log('allCookiesUser: acceptAllCookies!');
+  const cookieConsentModal = new ConsentModal();
+  await cookieConsentModal.isOpened();
+  await cookieConsentModal.clickAcceptAllCookies();
+};
+
+const userAcceptingAllCookies = Role(
+  getEnvUrl('/'),
+  async () => {
+    await acceptAllCookies();
+  },
+  { preserveUrl: false }
+);
```
There are still some issues with this. Sometimes, especially during the first run, the Testcafe miss-clicks the allow all cookies -button and instead of closing the cookie consent modal, it just minimizes it and then it blocks some clicks that we are expecting to happen during the tests.

terovirtanen commented:
## TestCafe result is __failed__ for https://liikunta-pr205.dev.hel.ninja :crying_cat_face::anger::boom::boom:

terovirtanen commented:
## TestCafe result is __failed__ for https://tapahtumat-pr205.dev.hel.ninja :crying_cat_face::anger::boom::boom:

terovirtanen commented:
## TestCafe result is __failed__ for https://liikunta-pr205.dev.hel.ninja :crying_cat_face::anger::boom::boom:

nenonja commented on packages/common-tests/browser-tests/page-model/landingPage.ts:
```diff
@@ -28,8 +28,9 @@ class LandingPage {
-    await t.expect(screen.findByText(this.searchText).exists).ok();
+    await t
+      .expect(screen.findByText(this.searchText).exists)
+      .ok({ timeout: 5000 });
```
I think that the adding general PageRequestTimeout from testcafe settings would be most robust way to fix this:
https://testcafe.io/documentation/402638/reference/configuration-file#pageRequestTimeout or 
https://testcafe.io/documentation/402638/reference/configuration-file#pageloadtimeout 

Hiccups are usually more related to the whole environment rather that single page,  so timeout settings could be set straight to the `yarn browser-test` CLI command because then `yarn browser-test:ci` command could have different (bigger) timeout values platta might have even more hiccups than local machine.

nikomakela replied:
this might be a wrong solution. I also tried `t.wait(5000)`, but the result was the same. This timeout is here for the debounce-feature and it just seems to take some time that the autocomplete answers appear.

nikomakela replied:
I don't think that using a global page request timeout is a solution here either. I think I'll just add some comments next to the timeout so it's more understandable.

nikomakela replied:
This is where it always fails 
![image](https://user-images.githubusercontent.com/389204/218418998-16d16c20-7a2f-4385-a6d8-84939fc3a29e.png)

This is what we should have
![image](https://user-images.githubusercontent.com/389204/218419299-012397ff-9eed-4634-b41f-9c159ac86311.png)

So, it seems like it has missed with the cookie consent modal, because it is minimized, not closed, but it also has missed to hit the search input (the autocomplete field).

nikomakela replied:
After using a t.wait() instead the timeout for ok, I also got this, where the menu has not yet appeared. 
<img width="1146" alt="image" src="https://user-images.githubusercontent.com/389204/218420824-bb36f2a9-cf66-417f-9d03-96cc877fd71f.png">

nikomakela replied:
the situation starts from this: 
![image](https://user-images.githubusercontent.com/389204/218445510-9f3d702d-82a8-4283-9f92-532aac5867a3.png)

The cookie consent modal is still open and the testcafe clicks the palceholder, so the modal gets minified.

nikomakela replied:
I refactored the initial "show cookie consent modal" state set. Now the modal should appear only if the cookie is not yet filled.

nenonja reviewed (changes requested):
Nice job! I found a one possible reason for flakyness which is not related to role but the selectors itself (see the comments). Currently all the test suites failed on platta, so this cannot be merged yet but I hope these problem will be fixed now. I started to wonder if the original problem was actually the selector but not the role in the first place? This could be tested too.... :)

nikomakela commented on packages/common-tests/browser-tests/page-model/eventSearchPage.ts:
```diff
@@ -3,11 +3,16 @@ import { t } from 'testcafe';
   private appNamespace: 'appHobbies' | 'appEvents' | 'appSports';
 
-  constructor(appNamespace: AppNamespace) {
+  constructor(appNamespace: AppNamespace | undefined = 'appEvents') {
```
I can recheck this, but the thing was that I had a common util that took advantage of this method. I can try to refactor it too or just give "any" app there in that code. I thought that it would be less confusing if the class constructor says that it's ok to use it without the appNamespace too.

nikomakela replied:
Okay, I'll change this so that the constructor allows only 1 of the 3 namespaces (as a mandatory) and I just use any of them in the common code.

nikomakela commented on packages/common-tests/browser-tests/page-model/consent-modal.ts:
```diff
@@ -1,45 +1,65 @@
   }
 
   public async isOpened() {
+    await t.expect(this.componentContainer.exists).ok();
```
You had another comment about whether the `isMinimized` is really needed or not. I would say that it is, because it really is one of the states that the consent can have open / minimized / closed. Earlier, I think we were just lucky, that when the cookie consent modal was minimised, it just wasn't in the way of other clicks. The minimized modal should not be considered as a closed one.

I need to have this componentContainer-check in isClosed-function, so wouldn't it make sense to have it in other state checks too?

nikomakela replied:
it's a super fast check anyway

nikomakela commented on packages/common-tests/browser-tests/page-model/consent-modal.ts:
```diff
@@ -1,45 +1,65 @@
+    console.log('ConsentModal: isOpened');
   }
+
+  public async isMinimized() {
```
Yea, it's currently not in use, but I could write a new test for it. I would say that it is needed, because it really is one of the states that the consent modal can have: open / minimized / closed.

nikomakela commented on packages/common-tests/browser-tests/page-model/eventDetailsPage.ts:
```diff
@@ -0,0 +1,151 @@
+import { initTestI18n as i18n } from '../../../common-i18n/src';
+
+class EventDetailsPage {
+  title = Selector('h1');
```
true that, I thought the same. Probably I add a new property to the current heading so it could be found in the tests. There is none at this point.

nikomakela commented on packages/common-tests/browser-tests/page-model/consent-modal.ts:
```diff
@@ -1,45 +1,65 @@
   private get title() {
-    return screen.findByRole('heading', {
-      name: `${this.appName} käyttää evästeitä`,
+    return screen.getByRole('heading', {
```
after changing all these, the result was not any better. The result didn't actually change at all.

terovirtanen commented:
**Events-Helsinki branch is deployed to platta: https://tapahtumat-pr205.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**Hobbies-Helsinki branch is deployed to platta: https://harrastukset-pr205.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**Sports-Helsinki branch is deployed to platta: https://liikunta-pr205.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
## TestCafe result is __failed__ for https://tapahtumat-pr205.dev.hel.ninja :crying_cat_face::anger::boom::boom:

terovirtanen commented:
## TestCafe result is __failed__ for https://harrastukset-pr205.dev.hel.ninja :crying_cat_face::anger::boom::boom:

terovirtanen commented:
**EVENTS GRAPHQL PROXY is deployed to platta: https://events-graphql-proxy-pr205.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
## TestCafe result is __success__ for https://liikunta-pr205.dev.hel.ninja :laughing::tada::tada::tada:

terovirtanen commented:
**VENUE GRAPHQL PROXY is deployed to platta: https://venue-graphql-proxy-pr205.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**Events-Helsinki branch is deployed to platta: https://tapahtumat-pr205.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**Sports-Helsinki branch is deployed to platta: https://liikunta-pr205.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**Hobbies-Helsinki branch is deployed to platta: https://harrastukset-pr205.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**EVENTS GRAPHQL PROXY is deployed to platta: https://events-graphql-proxy-pr205.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**VENUE GRAPHQL PROXY is deployed to platta: https://venue-graphql-proxy-pr205.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
## TestCafe result is __failed__ for https://tapahtumat-pr205.dev.hel.ninja :crying_cat_face::anger::boom::boom:

terovirtanen commented:
## TestCafe result is __success__ for https://liikunta-pr205.dev.hel.ninja :laughing::tada::tada::tada:

terovirtanen commented:
## TestCafe result is __failed__ for https://harrastukset-pr205.dev.hel.ninja :crying_cat_face::anger::boom::boom:

nikomakela committed:
refactor: show event cookie consent modal default state assignment

terovirtanen commented:
**Hobbies-Helsinki branch is deployed to platta: https://harrastukset-pr205.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**Events-Helsinki branch is deployed to platta: https://tapahtumat-pr205.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
## TestCafe result is __success__ for https://liikunta-pr205.dev.hel.ninja :laughing::tada::tada::tada:

terovirtanen commented:
**EVENTS GRAPHQL PROXY is deployed to platta: https://events-graphql-proxy-pr205.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**VENUE GRAPHQL PROXY is deployed to platta: https://venue-graphql-proxy-pr205.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
## TestCafe result is __failed__ for https://tapahtumat-pr205.dev.hel.ninja :crying_cat_face::anger::boom::boom:

terovirtanen commented:
## TestCafe result is __failed__ for https://tapahtumat-pr205.dev.hel.ninja :crying_cat_face::anger::boom::boom:

terovirtanen commented:
**Events-Helsinki branch is deployed to platta: https://tapahtumat-pr205.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**Sports-Helsinki branch is deployed to platta: https://liikunta-pr205.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**EVENTS GRAPHQL PROXY is deployed to platta: https://events-graphql-proxy-pr205.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**VENUE GRAPHQL PROXY is deployed to platta: https://venue-graphql-proxy-pr205.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
## TestCafe result is __failed__ for https://tapahtumat-pr205.dev.hel.ninja :crying_cat_face::anger::boom::boom:

terovirtanen commented:
## TestCafe result is __success__ for https://liikunta-pr205.dev.hel.ninja :laughing::tada::tada::tada:

terovirtanen commented:
## TestCafe result is __failed__ for https://harrastukset-pr205.dev.hel.ninja :crying_cat_face::anger::boom::boom:

terovirtanen commented:
## TestCafe result is __failed__ for https://tapahtumat-pr205.dev.hel.ninja :crying_cat_face::anger::boom::boom:

terovirtanen commented:
**Events-Helsinki branch is deployed to platta: https://tapahtumat-pr205.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**Hobbies-Helsinki branch is deployed to platta: https://harrastukset-pr205.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**Sports-Helsinki branch is deployed to platta: https://liikunta-pr205.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**EVENTS GRAPHQL PROXY is deployed to platta: https://events-graphql-proxy-pr205.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
## TestCafe result is __success__ for https://liikunta-pr205.dev.hel.ninja :laughing::tada::tada::tada:

terovirtanen commented:
**VENUE GRAPHQL PROXY is deployed to platta: https://venue-graphql-proxy-pr205.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
## TestCafe result is __failed__ for https://tapahtumat-pr205.dev.hel.ninja :crying_cat_face::anger::boom::boom:

terovirtanen commented:
**Hobbies-Helsinki branch is deployed to platta: https://harrastukset-pr205.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**Sports-Helsinki branch is deployed to platta: https://liikunta-pr205.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**VENUE GRAPHQL PROXY is deployed to platta: https://venue-graphql-proxy-pr205.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
## TestCafe result is __failed__ for https://harrastukset-pr205.dev.hel.ninja :crying_cat_face::anger::boom::boom:

terovirtanen commented:
## TestCafe result is __success__ for https://liikunta-pr205.dev.hel.ninja :laughing::tada::tada::tada:

terovirtanen commented:
**Hobbies-Helsinki branch is deployed to platta: https://harrastukset-pr205.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**Events-Helsinki branch is deployed to platta: https://tapahtumat-pr205.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**Sports-Helsinki branch is deployed to platta: https://liikunta-pr205.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
## TestCafe result is __failed__ for https://harrastukset-pr205.dev.hel.ninja :crying_cat_face::anger::boom::boom:

terovirtanen commented:
**VENUE GRAPHQL PROXY is deployed to platta: https://venue-graphql-proxy-pr205.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
## TestCafe result is __failed__ for https://tapahtumat-pr205.dev.hel.ninja :crying_cat_face::anger::boom::boom:

terovirtanen commented:
**EVENTS GRAPHQL PROXY is deployed to platta: https://events-graphql-proxy-pr205.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
## TestCafe result is __success__ for https://liikunta-pr205.dev.hel.ninja :laughing::tada::tada::tada:

terovirtanen commented:
## TestCafe result is __failed__ for https://tapahtumat-pr205.dev.hel.ninja :crying_cat_face::anger::boom::boom:

terovirtanen commented:
## TestCafe result is __failed__ for https://harrastukset-pr205.dev.hel.ninja :crying_cat_face::anger::boom::boom:

terovirtanen commented:
**Events-Helsinki branch is deployed to platta: https://tapahtumat-pr205.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**Hobbies-Helsinki branch is deployed to platta: https://harrastukset-pr205.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**Sports-Helsinki branch is deployed to platta: https://liikunta-pr205.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**VENUE GRAPHQL PROXY is deployed to platta: https://venue-graphql-proxy-pr205.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
## TestCafe result is __failed__ for https://tapahtumat-pr205.dev.hel.ninja :crying_cat_face::anger::boom::boom:

terovirtanen commented:
## TestCafe result is __failed__ for https://harrastukset-pr205.dev.hel.ninja :crying_cat_face::anger::boom::boom:

terovirtanen commented:
**EVENTS GRAPHQL PROXY is deployed to platta: https://events-graphql-proxy-pr205.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
## TestCafe result is __failed__ for https://liikunta-pr205.dev.hel.ninja :crying_cat_face::anger::boom::boom:

nikomakela committed:
test: remove the possible trailing slash from the browser tests base url

terovirtanen commented:
**Events-Helsinki branch is deployed to platta: https://tapahtumat-pr205.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**Hobbies-Helsinki branch is deployed to platta: https://harrastukset-pr205.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**Sports-Helsinki branch is deployed to platta: https://liikunta-pr205.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**EVENTS GRAPHQL PROXY is deployed to platta: https://events-graphql-proxy-pr205.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**VENUE GRAPHQL PROXY is deployed to platta: https://venue-graphql-proxy-pr205.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
## TestCafe result is __failed__ for https://liikunta-pr205.dev.hel.ninja :crying_cat_face::anger::boom::boom:

terovirtanen commented:
**Events-Helsinki branch is deployed to platta: https://tapahtumat-pr205.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**Hobbies-Helsinki branch is deployed to platta: https://harrastukset-pr205.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**Sports-Helsinki branch is deployed to platta: https://liikunta-pr205.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**EVENTS GRAPHQL PROXY is deployed to platta: https://events-graphql-proxy-pr205.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
## TestCafe result is __failed__ for https://tapahtumat-pr205.dev.hel.ninja :crying_cat_face::anger::boom::boom:

terovirtanen commented:
## TestCafe result is __failed__ for https://harrastukset-pr205.dev.hel.ninja :crying_cat_face::anger::boom::boom:

terovirtanen commented:
**VENUE GRAPHQL PROXY is deployed to platta: https://venue-graphql-proxy-pr205.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
## TestCafe result is __failed__ for https://liikunta-pr205.dev.hel.ninja :crying_cat_face::anger::boom::boom:

nikomakela committed:
test: inject the consent all cookie to every browser test

HH-294.
There has been plenty of issues with the use of Testcafe's useRole-feature.
The useRole -feature seems to work properly so that it is not executed multiple times,
but the cookie that it sets, is cleared quite randomly,
which makes the cookie consent modal to appear again and again.
When the cookie consent modal appears unexpectedly, the tests will fail!

As a solution to this, the cookie consent is set as a configuration
to every test that the Testcafe runs.

nikomakela committed:
test: navigation from event search page to details page and back

HH-294

terovirtanen commented:
**Hobbies-Helsinki branch is deployed to platta: https://harrastukset-pr205.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**Sports-Helsinki branch is deployed to platta: https://liikunta-pr205.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**Events-Helsinki branch is deployed to platta: https://tapahtumat-pr205.dev.hel.ninja** :rocket::rocket::rocket:

nikomakela commented:
There has been plenty of issues with the use of Testcafe's useRole-feature. The useRole -feature seems to work properly so that it is not executed multiple times, but the cookie that it sets, is cleared quite randomly, which makes the cookie consent modal to appear again and again. When the cookie consent modal appears unexpectedly, the tests will fail!

As a solution to this, the cookie consent is set as a configuration to every test that the Testcafe runs.

After all the suffering, I decided to switch off the whole cookie consent modal by injecting the cookie to every test. The allCookiesUser-Role is still there and it's also in the "best possible shape", so that we can start using it, if we find a way (or the Testcafe finds a way) to persist the cookies a bit longer.

terovirtanen commented:
**EVENTS GRAPHQL PROXY is deployed to platta: https://events-graphql-proxy-pr205.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
## TestCafe result is __success__ for https://harrastukset-pr205.dev.hel.ninja :laughing::tada::tada::tada:

terovirtanen commented:
## TestCafe result is __success__ for https://liikunta-pr205.dev.hel.ninja :laughing::tada::tada::tada:

terovirtanen commented:
**VENUE GRAPHQL PROXY is deployed to platta: https://venue-graphql-proxy-pr205.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
## TestCafe result is __success__ for https://tapahtumat-pr205.dev.hel.ninja :laughing::tada::tada::tada:

nenonja commented on apps/events-helsinki/.testcaferc.json:
```diff
@@ -1,6 +1,9 @@
-    { "module": "@testing-library/dom/dist/@testing-library/dom.umd.js" }
+    { "module": "@testing-library/dom/dist/@testing-library/dom.umd.js" },
+    {
+      "content": "document.cookie='city-of-helsinki-cookie-consents=%7B%22wordpress%22%3Atrue%2C%22linkedevents%22%3Atrue%2C%22matomo%22%3Atrue%2C%22i18next%22%3Atrue%2C%22city-of-helsinki-cookie-consents%22%3Atrue%2C%22servicemap_analytics%22%3Atrue%2C%22servicemap_session%22%3Atrue%2C%22servicemap_ga%22%3Atrue%7D; path=/;'"
```
The value is basically json object which is stringified and then url encoded. This could be more readable if the file would be renamed to a javascript file (testcaferc.js) and then value could be evaluated on the fly:
```
document.cookie='city-of-helsinki-cookie-consents=${{encodeURIComponent(JSON.stringify({"wordpress":true,"linkedevents":true,"matomo":true,"i18next":true,"city-of-helsinki-cookie-consents":true,"servicemap_analytics":true,"servicemap_session":true,"servicemap_ga":true}));
```

nenonja replied:
This is not necessaerily need to be done, but just would make the code more readable

nenonja commented on apps/sports-helsinki/src/domain/venue/similarVenues/__tests__/SimilarVenuesSection.test.tsx:
```diff
@@ -0,0 +1,107 @@
+  }),
+];
+
+const waitForComponentToBeLoaded = async () => {
```
I created similar kind of wait function but it's not in the main yet. This could be even more reboust with `{ timeout: 20_000 }`

terovirtanen commented:
**Events-Helsinki branch is deployed to platta: https://tapahtumat-pr205.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**Hobbies-Helsinki branch is deployed to platta: https://harrastukset-pr205.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**Sports-Helsinki branch is deployed to platta: https://liikunta-pr205.dev.hel.ninja** :rocket::rocket::rocket:

nikomakela committed:
test(sports): similar venues

LIIKUNTA-358

terovirtanen commented:
**EVENTS GRAPHQL PROXY is deployed to platta: https://events-graphql-proxy-pr205.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**VENUE GRAPHQL PROXY is deployed to platta: https://venue-graphql-proxy-pr205.dev.hel.ninja** :rocket::rocket::rocket:

nenonja reviewed (approved):
Looks good! Some comments and ideas, but could be merged finally, as well.

terovirtanen commented:
**Events-Helsinki branch is deployed to platta: https://tapahtumat-pr205.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**Hobbies-Helsinki branch is deployed to platta: https://harrastukset-pr205.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**Sports-Helsinki branch is deployed to platta: https://liikunta-pr205.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**VENUE GRAPHQL PROXY is deployed to platta: https://venue-graphql-proxy-pr205.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**EVENTS GRAPHQL PROXY is deployed to platta: https://events-graphql-proxy-pr205.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
## TestCafe result is __success__ for https://liikunta-pr205.dev.hel.ninja :laughing::tada::tada::tada:

terovirtanen commented:
## TestCafe result is __failed__ for https://tapahtumat-pr205.dev.hel.ninja :crying_cat_face::anger::boom::boom:

terovirtanen commented:
## TestCafe result is __success__ for https://harrastukset-pr205.dev.hel.ninja :laughing::tada::tada::tada:

terovirtanen commented:
## TestCafe result is __success__ for https://tapahtumat-pr205.dev.hel.ninja :laughing::tada::tada::tada:
